### ISSUE TITLE
Make chosen IP available in app scripts

### ISSUE THREAD

**Issue by twinj** [ionic-team/ionic-cli#2082]
**Note: for support questions, please use one of these channels:**

#### What version of the CLI are you using? Output of `ionic --version`
3 beta 5

#### Short description of the problem:
When running a live reload build I would like to be able to access the chosen Ip through the app scripts in my custom web pack config so that I can use it in my scripts, e.g: CORS policy changes in the index.html for different machines and environments.

`ionic cordova:emulate --target="iPad-Air-2, 10.3" --debug --livereload --webpack="custom"`

Currently the chosen ip is only passed to the caller of serve and not serve itself. The ionic cli then modifies the config.xml using that IP but does not make it available.

The chosen IP could be added to the CommandLineOptions object using the 'address' key.

**Comment by dwieeb** [comment]
@twinj `--address` works for me in the version I'm working with. I'm going to close this issue, but please post a comment or something if it's still not working when beta8 is released.

As an aside, are you on Ionic Worldwide? Do you mind if we chat? I'm `@dan` https://ionicworldwide.herokuapp.com/

**Comment by twinj** [comment]
This is a feature request aside from the usage of --address. It makes sense for the chosen ip from the given list to be passed to --address if a user decides to use this way. I use it this way because i don't want to hardcore the ip in my scripts. each environment or workstation has a different one. the code has already made the effort to give you a list, finish it off by making the chosen up available at runtime.

**Comment by dwieeb** [comment]
Sorry, I see what you're saying now. If the interactive list pops up, it's not getting the value from it. Should be fixed for the next beta! 

Thanks!

**Comment by jthoms1** [comment]
Actually it shouldn't show the interactive list if address is supplied.

https://github.com/driftyco/ionic-cli/blob/v3/packages/cli-plugin-ionic-angular/src/serve.ts#L19

**Comment by twinj** [comment]
The address is not being supplied. The chosen address from the list should be passed to the app scripts. It is not.

**Comment by twinj** [comment]
Is there a misunderstanding still with what is happening here? I created a pull request to fix this issue and it was closed. Citing reason was that updating the address option breaks the app script contract?

It fulfils it in my opinion.

As a user if I run the command:

`ionic cordova emulate -l --address 10.10.10.10`

All runs as expected. The code first checks that the given address exists in the list of interfaces. If it does not exist the code then shows that list and the user chooses one. In the instance where the original supplied address exists it is passed to app script serve as well as updating the chosenIp variable which determines the config.xml setup. If, however, I have to choose address the address it is not updated and not passed to the app script serve and so only updates the config.xml. This has created problems for me in that I use this address in my index html for CORS.

If I run the command without using the address option and there are more than 1 interface available
I get the same behaviour as an incorrect address as explained above.

What is the expected behaviour here? Is the first case wrong and the address, when correct as passed by --address, should not be passed to the app scripts serve or is it that it should? 

If the address option name clashes with app scripts then change it so as to NOT break that contract. If it is the same then it MUST be passed and updated if a new one is chosen.

From a user perspective using the --address option and selecting the address manually SHOULD have the same behaviour as they are in essence selecting the same option. They currently do not. 

This is not an enhancement but a bug.
